bin/check-instance-events.rb

```diff
@@ -66,55 +66,68 @@ def aws_config
       region: config[:aws_region] }
   end
 
+  def ec2_regions
+    # This is for SDK v2
+    # Aws.partition('aws').regions.map(&:name)
+
+    AWS::EC2.regions.map(&:name)
+  end
+
   def run
     event_instances = []
     aws_config = {}
 
+    aws_regions = ec2_regions
+
+    unless config[:aws_region].casecmp('all') == 0
```

Comment:
sdk v2 is always prefered, but we are not requiring existing scripts to be rewritten for sdk v2.  You are right that sdk v1 scripts that don't use the common code would not break.  So in this case it should be fine.  Certainly if you have the time to refactor for sdk v2 that would be excellent.